envbuilder.go

```diff
@@ -354,8 +373,10 @@ ENTRYPOINT [%q]`, exePath, exePath, exePath)
 	stderrWriter, closeStderr := log.Writer(opts.Logger)
 	defer closeStderr()
 	build := func() (v1.Image, error) {
-		_, err := opts.Filesystem.Stat(constants.MagicFile)
-		if err == nil && opts.SkipRebuild {
+		defer cleanupBuildContext()
```

Comment:
I didn't consider this before, but this cleanup made me think, should we try hard(er) to make sure we don't accidentally conflict with any existing files in the buildcontext? Say, if we place these under `.envbuilder.tmp/envbuilder` and `.envbuilder.tmp/image`?

Incidentally, if the build context is R/O we're also in for a bad time but let's defer that problem 😄.

Reply:
I like `.envbuilder.tmp`, especially for https://github.com/coder/kaniko/pull/19